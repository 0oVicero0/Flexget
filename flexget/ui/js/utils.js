'use strict';

String.prototype.startsWith = function(prefix) {
  return this.indexOf(prefix) === 0;
};

function registerFlexModule(module) {
  angular.module('flexgetApp').requires.push(module.name);
}

app.menu = [];
app.routes = [];

function register_menu(href, caption, icon, order) {
  href = '/ui/#' + href;
  app.menu.push({href: href, caption: caption, icon: icon, order: order})
}

function register_route(name, url, controller, template) {
  app.routes.push({
    name: name,
    url: url,
    controller: controller,
    template: template
  });
}