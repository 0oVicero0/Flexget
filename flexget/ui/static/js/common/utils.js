'use strict';

String.prototype.startsWith = function(prefix) {
    return this.indexOf(prefix) === 0;
};

function registerFlexModule(module) {
  angular.module('flexgetApp').requires.push(module.name);
}