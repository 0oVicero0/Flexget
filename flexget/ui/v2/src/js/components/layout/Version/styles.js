import styled, { css } from 'emotion/react';
import theme from 'theme';

export const version = css`color: ${theme.palette.error[500]}`;
export const Wrapper = styled.div`
  composes: ${version};
`;

export const Line = styled.p`
  margin: 0;
`;
