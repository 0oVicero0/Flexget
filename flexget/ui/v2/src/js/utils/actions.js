export const createAction = namespace => (type, payload, {
  ignore = false,
  message,
} = {}) => ({
  type,
  payload,
  error: (payload instanceof Error),
  meta: {
    ignore,
    message,
    namespace,
  },
});

export const createLoading = namespace => type => ({
  type,
  meta: {
    namespace,
    loading: true,
  },
});
