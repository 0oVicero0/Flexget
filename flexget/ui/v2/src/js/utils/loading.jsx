import Loadable from 'react-loadable';
import Loading from 'components/common/Loader';

export function createAsyncComponent(loader) {
  return Loadable({
    loading: Loading,
    loader,
  });
}

