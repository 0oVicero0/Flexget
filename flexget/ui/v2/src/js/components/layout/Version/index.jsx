import React, { Component } from 'react';
import PropTypes from 'prop-types';
import semver from 'semver-compare';
import IconButton from 'material-ui/IconButton';
import Icon from 'material-ui/Icon';
import { withStyles, createStyleSheet } from 'material-ui/styles';
import 'font-awesome/css/font-awesome.css';

const styleSheet = createStyleSheet('Version', theme => ({
  version: {
    color: theme.palette.error[500],
  },
  par: {
    margin: 0,
  },
}));

class Version extends Component {
  static propTypes = {
    classes: PropTypes.object.isRequired,
    version: PropTypes.shape({
      api: PropTypes.string,
      flexget: PropTypes.string,
      latest: PropTypes.string,
    }).isRequired,
    getVersion: PropTypes.func.isRequired,
  };

  componentDidMount() {
    this.props.getVersion();
  }

  render() {
    const { classes, version: { api, flexget, latest } } = this.props;
    return (
      <div className={classes.version}>
        <p className={classes.par}>Version Info</p>
        <p className={classes.par}>Flexget: { flexget } {
          latest && semver(latest, flexget) === 1 && (
            <IconButton href="https://flexget.com/ChangeLog">
              <Icon className={`fa fa-question-circle-o ${classes.version}`} />
            </IconButton>
          ) } </p>
        <p className={classes.par}>API: { api }</p>
      </div>
    );
  }
}

export default withStyles(styleSheet)(Version);
