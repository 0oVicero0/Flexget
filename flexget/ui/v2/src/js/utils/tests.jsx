import React from 'react';
import { MuiThemeProvider } from 'material-ui/styles';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';

const mockStore = configureMockStore();

export function themed(component) {
  return (
    <MuiThemeProvider>
      { component }
    </MuiThemeProvider>
  );
}

export function router(component) {
  return (
    <MemoryRouter>
      { component }
    </MemoryRouter>
  );
}

export function provider(component, state = {}) {
  return (
    <Provider store={mockStore(state)}>
      { component }
    </Provider>
  );
}
