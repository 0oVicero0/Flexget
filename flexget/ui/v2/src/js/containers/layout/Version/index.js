import { connect } from 'react-redux';
import { GET_VERSION } from 'actions/version';
import { request } from 'utils/actions';
import Version from 'components/layout/Version';

export function mapStateToProps({ version }) {
  return {
    version,
  };
}

function mapDispatchToProps(dispatch) {
  return {
    getVersion: () => dispatch(request(GET_VERSION)),
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(Version);
