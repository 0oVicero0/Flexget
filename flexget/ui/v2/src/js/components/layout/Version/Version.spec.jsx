import React from 'react';
import renderer from 'react-test-renderer';
import Version from 'components/layout/Version';
import { themed } from 'utils/tests';

describe('components/layout/Version', () => {
  it('renders correctly with latest version', () => {
    const getVersion = jest.fn();
    const tree = renderer.create(
      themed(<Version version={{ api: '1.1.2', flexget: '2.10.60', latest: '2.10.60' }} getVersion={getVersion} />),
    ).toJSON();
    expect(tree).toMatchSnapshot();
    expect(getVersion).toHaveBeenCalled();
  });

  it('renders correctly without latest version', () => {
    const getVersion = jest.fn();
    const tree = renderer.create(
      themed(<Version version={{ api: '1.1.2', flexget: '2.10.11', latest: '2.10.60' }} getVersion={getVersion} />),
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
