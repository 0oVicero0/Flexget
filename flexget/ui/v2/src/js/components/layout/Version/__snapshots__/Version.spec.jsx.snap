// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/layout/Version renders correctly with latest version 1`] = `
.glamor-7 {
  color: #f44336;
}

.glamor-1 {
  margin: 0px;
}

<div
  className="glamor-6 glamor-7"
>
  <p
    className="glamor-0 glamor-1"
  >
    Version Info
  </p>
  <p
    className="glamor-0 glamor-1"
  >
    Flexget: 
    2.10.60
     
     
  </p>
  <p
    className="glamor-0 glamor-1"
  >
    API: 
    1.1.2
  </p>
</div>
`;

exports[`components/layout/Version renders correctly without latest version 1`] = `
.glamor-2 {
  color: #f44336;
}

.glamor-1 {
  margin: 0px;
}

<div
  className="glamor-7 glamor-2"
>
  <p
    className="glamor-0 glamor-1"
  >
    Version Info
  </p>
  <p
    className="glamor-0 glamor-1"
  >
    Flexget: 
    2.10.11
     
    <a
      className="MuiButtonBase-root-10 MuiIconButton-root-1"
      href="https://flexget.com/ChangeLog"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
    >
      <span
        className="MuiIconButton-label-7"
      >
        <span
          aria-hidden="true"
          className="material-icons MuiIcon-root-12 MuiIconButton-icon-8 fa fa-question-circle-o glamor-2"
        />
      </span>
      <span
        className="MuiTouchRipple-root-19"
      />
    </a>
     
  </p>
  <p
    className="glamor-0 glamor-1"
  >
    API: 
    1.1.2
  </p>
</div>
`;
